pkg/storage/image.go

```diff
@@ -598,6 +603,25 @@ func GetImageService(ctx context.Context, sc *types.SystemContext, store storage
 		imageCache:            make(map[string]imageCacheItem),
 		ctx:                   ctx,
 	}
+
+	sysregistries, err := sysregistriesv2.GetRegistries(sc)
+	if err != nil {
+		return nil, err
+	}
+	blockedRegistries := make(map[string]bool)
+	unqualifiedSearchRegistries := []string{}
+	for _, r := range sysregistries {
+		if r.Blocked {
+			blockedRegistries[r.URL] = true
+			continue
+		}
+		if r.Search {
+			unqualifiedSearchRegistries = append(unqualifiedSearchRegistries, r.URL)
```

Comment:
It’s not obviously true that `GetRegistries` returns search entries in the user-intended order. Why is this an integral part of the “blocked” feature?